### PR TITLE
[WIP, low priority] Update sharing to show icon on iOS

### PR DESCRIPTION
**Pull request by AlanSl**
For https://github.com/project-vagabond/covid-green-app/issues/36

Lower priority, needs more testing to make sure the new dependency doesn't cause problems in older iOS etc. I'm not expecting this to go in before the testing and it might be a won't-fix.

I'm limited in what I can test on iOS simulator - in particular I can't test how apps like mail and messages actually pick this up. There are lots of app-type specific options in `react-native-share`, docs at https://react-native-community.github.io/react-native-share/ so feel free to push any fixes e.g. for specific app types. We might also want to exclude some app types.

Screenshot 2020-08-31 at 11 41 59

### CODE REVIEW

**Review by segfault — APPROVED** [review]
LGTM. 👍

File: src/components/atoms/navbar.tsx [hunk 1]
```diff
@@ -23,19 +24,38 @@ interface NavBarProps {
...
+
   try {
-    await Share.share(
-      {
-        title: t('common:message'),
-        message:
-          Platform.OS === 'android' ? t('common:url') : t('common:message'),
-        url: t('common:url')
+    await Share.open(Platform.select({
+      ios: {
```
Comment (davidjharding) [inline src/components/atoms/navbar.tsx]: What about Android?

File: src/components/atoms/navbar.tsx [hunk 2]
```diff
@@ -23,19 +24,38 @@ interface NavBarProps {
...
+            },
+            linkMetadata: {
+               title: title,
+               icon: icon
+            }
+          },
+        ],
+      },
+      default: {
+        title: 'abc',
```
Comment (davidjharding) [inline src/components/atoms/navbar.tsx]: Shouldn't the default title be `title` and not `"abc"`

**Review by dharding — CHANGES_REQUESTED** [review]

**Review by dharding — CHANGES_REQUESTED** [review]
Also, did you test all of the extra options such as save to image and all? If not, perhaps we should remove these and focus on the basics.

File: src/components/atoms/navbar.tsx (hunk 2 above)
Comment (AlanSl) [reply src/components/atoms/navbar.tsx]: You're right, it was a placeholder during testing, I just posted this to get some more in device testing

**Comment by AlanSl** [comment]
@dharding Thanks for the comments, I said in the description that I wasn't intending to get this in today and it's lower priority - I just wanted to open it up for wider in-device testing of the basics to make sure there are no negative side effects of the new dependency.

Sorry if that wasn't clear from the description, I've switched this to `draft` to be clearer.

File: src/components/atoms/navbar.tsx (hunk 1 above)
Comment (AlanSl) [reply src/components/atoms/navbar.tsx]: That's caught by `default` case below

**Comment by AlanSl** [comment]
Closing because these issues are no longer on the launch board and the branches are out of date

**Pull request closed without merging**